beacon-chain/core/state/state.go

```diff
@@ -5,45 +5,17 @@ package state
 
 import (
 	"bytes"
-	"encoding/binary"
 	"fmt"
 
+	"github.com/prysmaticlabs/go-ssz"
 	b "github.com/prysmaticlabs/prysm/beacon-chain/core/blocks"
 	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
-	v "github.com/prysmaticlabs/prysm/beacon-chain/core/validators"
 	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
-	"github.com/prysmaticlabs/prysm/shared/hashutil"
 	"github.com/prysmaticlabs/prysm/shared/params"
 )
 
-// GenesisBeaconState gets called when DepositsForChainStart count of
-// full deposits were made to the deposit contract and the ChainStart log gets emitted.
-//
-// Spec pseudocode definition:
-//  def get_genesis_beacon_state(deposits: List[Deposit],
-//                             genesis_time: int,
-//                             genesis_eth1_data: Eth1Data) -> BeaconState:
-//    """
-//    Get the genesis ``BeaconState``.
-//    """
-//    state = BeaconState(genesis_time=genesis_time, latest_eth1_data=genesis_eth1_data)
-//
-//    # Process genesis deposits
-//    for deposit in genesis_validator_deposits:
-//        process_deposit(state, deposit)
-//
-//    # Process genesis activations
-//    for validator in state.validator_registry:
-//        if validator.effective_balance >= MAX_EFFECTIVE_BALANCE:
-//            validator.activation_eligibility_epoch = GENESIS_EPOCH
-//            validator.activation_epoch = GENESIS_EPOCH
-//
-//    genesis_active_index_root = hash_tree_root(get_active_validator_indices(state, GENESIS_EPOCH))
-//    for index in range(LATEST_ACTIVE_INDEX_ROOTS_LENGTH):
-//        state.latest_active_index_roots[index] = genesis_active_index_root
-//
-//    return state
-func GenesisBeaconState(deposits []*pb.Deposit, genesisTime uint64, eth1Data *pb.Eth1Data) (*pb.BeaconState, error) {
+// BeaconState gets called to return a beacon state where all the fields are set to genesis values.
+func BeaconState(blkHeader *pb.BeaconBlockHeader, genesisTime uint64, eth1Data *pb.Eth1Data) *pb.BeaconState {
```

Comment:
what's the difference between `BeaconState` and `GenesisBeaconState`? Why aren't they together? We only need to call `BeaconState` once

Reply:
it becomes a vary long function if we don't separate the initialization from the computation

Reply:
I am not sure what you mean by this, why cant we do it the same way we have been doing as before ?

Reply:
removed